app/components/category-item.js

```diff
@@ -1,47 +1,121 @@
 import Ember from 'ember';
 
-export default Ember.Component.extend({
+const {
+  Component,
+  computed,
+  get,
+  inject: {
+    service,
+  },
+  set,
+} = Ember;
+
+
+/**
+ * `category-item` composes a category's icon, and description.
+ *
+ * ## default usage
+ *
+ * ```handlebars
+ * {{category-item category=category}}
+ * ```
+ *
+ * @class category-item
+ * @module Component
+ * @extends Ember.Component
+ */
+export default Component.extend({
   classNames: ['category-item'],
   classNameBindings: ['selected'],
   isLoading: false,
 
-  flashMessages: Ember.inject.service(),
-  userCategories: Ember.inject.service(),
+  /**
+   * @property flashMessages
+   * @type Ember.Service
+   */
+  flashMessages: service(),
+
+  /**
+   * @property userCategories
+   * @type Ember.Service
+   */
+  userCategories: service(),
 
-  selected: Ember.computed.notEmpty('userCategory'),
+  /**
+   * Returns if the category has been selected by the user.
+   *
+   * @property selected
+   * @type Boolean
+   */
+  selected: computed.notEmpty('userCategory'),
+
+  /**
+   * Fetches the category using the `userCategories` service.
+   *
+   * @property userCategory
+   * @type {Object} category
+   */
+  userCategory: computed('category', 'userCategories.userCategories', function() {
+    const category = get(this, 'category');
+    const userCategories = get(this, 'userCategories');
 
-  userCategory: Ember.computed('category', 'userCategories.userCategories', function() {
-    let category = this.get('category');
-    let userCategories = this.get('userCategories');
     return userCategories.findUserCategory(category);
   }),
 
   actions: {
+
+    /**
+     * Action that sets the `isLoading` property to `true` and saves the
+     * category to the user's categories. If there is an error saving the
+     * category, a flash message is displayed. The `isLoading` property is
+     * reset after the save passes/fails.
+     *
+     * @method addCategory
+     * @param {Object} category
+     */
     addCategory(category) {
-      this.set('isLoading', true);
-      let userCategories = this.get('userCategories');
+      set(this, 'isLoading', true);
+      const userCategories = get(this, 'userCategories');
+
       return userCategories.addCategory(category).catch(() => {
-        let message = `An error occurred trying to add ${category.get('name')}.`;
+        const message = `An error occurred trying to add ${category.get('name')}.`;
```

Comment:
This may be opening a can of worms, but have we landed on a general policy around defaulting to `const` vs `let`? I used to be a "constantly const" person but ever since I learned that [@rwjblue subscribes to the "liberal let" philosophy](http://discuss.emberjs.com/t/what-is-the-difference-between-const-let-var/9559/3?u=tomchen), I've default to `let`. Again, I'm not dead set on this...